benchmarks/python/conftest.py

```diff
@@ -28,6 +28,20 @@ def pytest_addoption(parser):
         help="Benchmarks torch.compile mode.",
     )
 
+    parser.addoption(
+        "--rounds",
+        action="store",
+        default=10,
+        help="Number of rounds for each benchmark.",
+    )
+
+    parser.addoption(
+        "--warmup-rounds",
+        action="store",
+        default=1,
+        help="Number of warmup rounds for each benchmark.",
+    )
+
```

Comment:
Looks like pytest-benchmark itself has some related options:

https://pytest-benchmark.readthedocs.io/en/latest/usage.html#commandline-options

Do we need these our own options? Also, pytest-benchmark uses "rounds" and "iterations", and I'm not sure what the difference is.

Reply:
There is no command line option to control the `rounds/iterations` in the pedantic mode (https://pytest-benchmark.readthedocs.io/en/latest/pedantic.html). They are passed directly to `benchmark.pedantic`. These CLI options are essentially a utility to update those numbers. 

Each round has a given number of iterations. When we use a setup function (which is our case to clear L2 cache), iterations=1.

Reply:
Thanks. What do the round and the iteration mean?

Reply:
We execute a given target function for `n` rounds with each round itself consists of `k` iterations -- that is within each round, the target function is executed `k` times. So in total, we execute the benchmark `nk`. The statistics of the benchmarks are based on the duration of these rounds (mean, median..).  The motivation behind iterations seem to be able to measure functions that have smaller duration than the precision of the timer.

Reference: https://github.com/ionelmc/pytest-benchmark/issues/186

Reply:
Can you please leave a comment that the default pytest-benchmark options are not available as we use the pedantic mode?